src/main/java/org/reactome/release/dataexport/configuration/ConfigurationManager.java

```diff
@@ -235,9 +242,23 @@ private String getConfigurationKeyFromConfigurationEntry(String configurationEnt
 	}
 
 	private void makeFileReadAndWriteForUserAndGroupOnly() throws IOException {
+		if (SystemUtils.IS_OS_WINDOWS) {
+			giveFileReadAndWritePermissionsForUserInWindows();
+		} else {
+			giveFileReadAndWritePermissionsForUserAndGroupOnlyInUnix();
+		}
+	}
+
+	private void giveFileReadAndWritePermissionsForUserInWindows() {
+		File configFile = new File(getConfigFileName());
+		configFile.setReadable(true);
+		configFile.setWritable(true);
```

Comment:
It writes to the config file because it is part of the program instead (if no proper config file already exists) of being set up by a shell script.  It was easier and made more sense to verify the configuration in the program because the program knows what it needs.  Creating a configuration file was an extension of that so that configuration information was kept in one place instead of repeated in a shell script that could get desynced.

This particular line, though, just sets the permissions for the file as writable so there's no permission errors but doesn't actually change the file at this point.

Reply:
Ok. It just seems strange to do this since all of our other programs only read configurations. If it's writing a config... why bother writing it? It already has the correct values. The config files are usually used to pass in values that could vary. If this program knows what should be in a config, why even bother with a config?

Reply:
The program will only read the configuration file if it is already present and has the needed values and won't write anything.  If it detects there is no configuration file or missing values are needed, the user will be prompted to provide the values and a file is written.  It's just a check to ensure a proper configuration exists and is correct and only write a new one with user input if not.